linera-execution/solidity/LineraTypes.sol

```diff
@@ -44,292 +43,313 @@ library LineraTypes {
     }
 
-    function bcs_deserialize_OptionBool(bytes memory input)
+    function bcs_deserialize_AccountOwner(bytes memory input)
         internal
         pure
-        returns (OptionBool)
+        returns (AccountOwner memory)
     {
         uint256 new_pos;
-        OptionBool value;
-        (new_pos, value) = bcs_deserialize_offset_OptionBool(0, input);
+        AccountOwner memory value;
+        (new_pos, value) = bcs_deserialize_offset_AccountOwner(0, input);
         require(new_pos == input.length, "incomplete deserialization");
         return value;
     }
 
-    struct TimeDelta {
-        uint64 value;
+    struct ApplicationId {
+        CryptoHash application_description_hash;
     }
 
-    function bcs_serialize_TimeDelta(TimeDelta memory input)
+    function bcs_serialize_ApplicationId(ApplicationId memory input)
         internal
         pure
         returns (bytes memory)
     {
-        bytes memory result = bcs_serialize_uint64(input.value);
-        return result;
+        return bcs_serialize_CryptoHash(input.application_description_hash);
     }
 
-    function bcs_deserialize_offset_TimeDelta(uint256 pos, bytes memory input)
+    function bcs_deserialize_offset_ApplicationId(uint256 pos, bytes memory input)
         internal
         pure
-        returns (uint256, TimeDelta memory)
+        returns (uint256, ApplicationId memory)
     {
-        uint256 new_pos = pos;
-        uint64 value;
-        (new_pos, value) = bcs_deserialize_offset_uint64(new_pos, input);
-        return (new_pos, TimeDelta(value));
+        uint256 new_pos;
+        CryptoHash memory application_description_hash;
+        (new_pos, application_description_hash) = bcs_deserialize_offset_CryptoHash(pos, input);
+        return (new_pos, ApplicationId(application_description_hash));
     }
 
-    function bcs_deserialize_TimeDelta(bytes memory input)
+    function bcs_deserialize_ApplicationId(bytes memory input)
         internal
         pure
-        returns (TimeDelta memory)
+        returns (ApplicationId memory)
     {
         uint256 new_pos;
-        TimeDelta memory value;
-        (new_pos, value) = bcs_deserialize_offset_TimeDelta(0, input);
+        ApplicationId memory value;
+        (new_pos, value) = bcs_deserialize_offset_ApplicationId(0, input);
         require(new_pos == input.length, "incomplete deserialization");
         return value;
     }
 
-    struct key_values_AccountOwner_uint64 {
-        AccountOwner key;
-        uint64 value;
+    struct BaseRuntimePrecompile {
+        uint8 choice;
+        // choice=0 corresponds to ChainId
+        // choice=1 corresponds to ApplicationCreatorChainId
+        // choice=2 corresponds to ChainOwnership
+        // choice=3 corresponds to ReadDataBlob
+        BaseRuntimePrecompile_ReadDataBlob read_data_blob;
+        // choice=4 corresponds to AssertDataBlobExists
+        BaseRuntimePrecompile_AssertDataBlobExists assert_data_blob_exists;
```

Comment:
Those have the same field; would it make sense to just have a single `CryptoHash` field instead, so we don't waste memory?

Reply:
Yes, but that would have to be coded in `serde-reflection` and the coding of the API would have to be done.